lib/basedriver/helpers.js

```diff
@@ -357,6 +357,27 @@ function duplicateKeys (input, firstKey, secondKey) {
   return input;
 }
 
+/**
+ * Takes a desired capability and tries to JSON.parse it as an array,
+ * and either returns the parsed array or a singleton array.
+ *
+ * @param {string|Array<String>} cap A desired capability
+ */
+function parseCapsArray (cap) {
+  let parsedCaps;
+  try {
+    parsedCaps = JSON.parse(cap);
+  } catch (ign) { }
```

Comment:
Could you add a comment here why we are ignoring the exception? I understood your reason after some time but it would be good to have it documented here

Reply:
@umutuzgur take a look at this new approach for re review